File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var findThePrefixCommonArray = function (A, B) {
    let lenA = Math.max(...A)
    let freA = new Array(lenA + 1).fill(0)
    let lenB = Math.max(...B)
    let freB = new Array(lenA + 1).fill(0)
    let res = []
    for (let i = 0; i < A.length; i++) {
        freA[A[i]] += 1
        freB[B[i]] += 1
        let max = Math.max(A[i], B[i])
        let count = 0
        for (let j = 0; j < freA.length; j++) {
            if (freA[j] >= 1 && freB[j] >= 1) {
                count++
            }
        }
        res.push(count)
    }
    return res
};

File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        lenA = max(A)
        freA = [0] * (lenA + 1)
        lenB = max(B)
        freB = [0] * (lenA + 1)
        res = []

        for i in range(len(A)):
            freA[A[i]] += 1
            freB[B[i]] += 1
            max_val = max(A[i], B[i])
            count = 0
            for j in range(len(freA)):
                if freA[j] >= 1 and freB[j] >= 1:
                    count += 1
            res.append(count)

        return res